tracer/src/Datadog.Trace/Propagators/SpanContextPropagator.cs

```diff
@@ -206,14 +216,28 @@ cumulativeSpanContext is not null &&
                 if (cumulativeSpanContext == null)
                 {
                     cumulativeSpanContext = currentExtractedContext.SpanContext;
+                    initialExtractorDisplayName = extractor.DisplayName;
                 }
-                else if (extractor is W3CTraceContextPropagator && currentExtractedContext.SpanContext is { } extractedSpanContext)
+                else if (currentExtractedContext.SpanContext is { } extractedSpanContext)
                 {
-                    MergeExtractedW3CSpanContext(cumulativeSpanContext, extractedSpanContext);
+                    if (cumulativeSpanContext.RawTraceId != extractedSpanContext.RawTraceId)
+                    {
+                        spanLinks.Add(new SpanLink(extractedSpanContext, attributes: [new("reason", "terminated_context"), new("context_headers", extractor.DisplayName)]));
```

Comment:
For each of the span attributes I want to get the name of the propagator used during the extraction, so I resorted to adding a `DisplayName` property. Is there a better approach for getting that info?

Reply:
I think you're right that we could rely on the type, maybe do some sort of switch statement/expression based on the type (instead of a property), though I want to make sure we show the user a name that maps to the config, such as `DD_TRACE_PROPAGATION_STYLE=tracecontext`. If we keep that end result in mind, I'm open to improvements/suggestions!